agent/engine/serviceconnect/version_linux.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package serviceconnect
+
+import (
+	"sort"
+)
+
+const (
+	appnetInterfaceV1          = "v1"
+	serviceConnectCapabilityV1 = "ecs.service-connect.v1"
+)
+
+var (
+	supportedAppnetInterfaceVerToCapability = map[string][]string{
+		appnetInterfaceV1: {
+			serviceConnectCapabilityV1,
+		},
+	}
+)
+
+// getSupportedAppnetInterfaceVersions returns the all the supported AppNet interface versions
+// by ECS Agent from highest to lowest version
+func getSupportedAppnetInterfaceVersions() []string {
+	supportedAppnetInterfaceVersions := make([]string, 0, len(supportedAppnetInterfaceVerToCapability))
+	for supportedAppnetInterfaceVersion := range supportedAppnetInterfaceVerToCapability {
+		supportedAppnetInterfaceVersions = append(supportedAppnetInterfaceVersions, supportedAppnetInterfaceVersion)
+	}
+	sort.Sort(sort.Reverse(sort.StringSlice(supportedAppnetInterfaceVersions)))
```

Comment:
Will this become problematic if we are comparing `v10` to `v2`?

Reply:
Good catch! I have updated the sorting logic to handle it correctly.
Thank you!